### PR TITLE
add binlog proto

### PR DESCRIPTION
**Pull request by coocood**
@shenli @siddontang @qiuyesuifeng @iamxy PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
why not in kvproto? 
@disksing

**Comment by coocood** [comment]
@siddontang binlog is SQL layer information, kvproto should not know anything about SQL.

**Comment by disksing** [comment]
@siddontang  tikv does not care about what is inside binlog. So in kvproto we can just use a `bytes`.

File: proto/binlog/binlog.proto [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import "gogoproto/gogo.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+// TableMutation contains mutations in a table.
+message TableMutation {
+    optional int64 tableID = 1 [(gogoproto.nullable) = false];
+    repeated int64 insertedIDs = 2;
```
Comment (qiuyesuifeng) [inline proto/binlog/binlog.proto]: Alignment?

Comment (qiuyesuifeng) [inline proto/binlog/binlog.proto]: I see kvproto use unix format, how about using `inserte_ids`?

File: proto/binlog/binlog.proto [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import "gogoproto/gogo.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+// TableMutation contains mutations in a table.
+message TableMutation {
+    optional int64 tableID = 1 [(gogoproto.nullable) = false];
```
Comment (siddontang) [inline proto/binlog/binlog.proto]: keep alignment for field number.

File: generate-binlog.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+cd proto/binlog
+
+echo "generate binlog code..."
+GOGO_ROOT=${GOPATH}/src/github.com/gogo/protobuf
+protoc -I.:${GOGO_ROOT}:${GOGO_ROOT}/protobuf --gofast_out=../../go-binlog *.proto
+cd ../../go-binlog
```
Comment (qiuyesuifeng) [inline generate-binlog.sh]: How about using `binlog` package?

Comment (coocood) [inline generate-binlog.sh]: 'go-binlog' follows the pattern of 'go-tipb'

**Comment by coocood** [comment]
@qiuyesuifeng @siddontang PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**